gravitee-apim-portal-webui-next/src/app/registration/registration.component.ts

```diff
@@ -0,0 +1,116 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, DestroyRef, OnInit, signal } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatButtonModule } from '@angular/material/button';
+import { MatCard, MatCardContent, MatCardHeader, MatCardTitle } from '@angular/material/card';
+import { MatIcon } from '@angular/material/icon';
+import { MatError, MatFormField, MatInput, MatLabel } from '@angular/material/input';
+import { MatOption, MatSelect } from '@angular/material/select';
+import { RouterLink } from '@angular/router';
+import { tap } from 'rxjs';
+
+import { MobileClassDirective } from '../../directives/mobile-class.directive';
+import { CustomUserFields } from '../../entities/user/custom-user-fields';
+import { UsersService } from '../../services/users.service';
+
+interface UserRegistrationFormValue {
+  firstname: string;
+  lastname: string;
+  email: string;
+  customFields?: { [key: string]: string };
+}
+
+@Component({
+  selector: 'app-registration',
+  imports: [
+    MatCard,
+    MatCardContent,
+    MatCardHeader,
+    MatCardTitle,
+    MatError,
+    MatFormField,
+    MatInput,
+    MatLabel,
+    MatButtonModule,
+    ReactiveFormsModule,
+    RouterLink,
+    MobileClassDirective,
+    MatError,
+    MatFormField,
+    MatOption,
+    MatSelect,
+    MatIcon,
+  ],
+  templateUrl: './registration.component.html',
+  styleUrl: './registration.component.scss',
+})
+export class RegistrationComponent implements OnInit {
+  submitted: boolean = false;
+  sentToEmail = '';
+
+  customUserFields: CustomUserFields[] = [];
+
+  registrationForm: FormGroup<{
+    firstname: FormControl;
+    lastname: FormControl;
+    email: FormControl;
+    customFields?: FormGroup;
+  }> = new FormGroup({
+    firstname: new FormControl('', [Validators.required]),
+    lastname: new FormControl('', [Validators.required]),
+    email: new FormControl('', [Validators.required, Validators.email]),
+  });
+
+  error = signal(200);
+
+  constructor(
+    private usersService: UsersService,
+    private readonly destroyRef: DestroyRef,
+  ) {}
+
+  ngOnInit(): void {
+    this.usersService
+      .listCustomUserFields()
+      .pipe(
+        tap(customUserFields => {
+          this.customUserFields = customUserFields;
```

Comment:
Related to https://github.com/gravitee-io/gravitee-api-management/pull/14707/changes#r2640097808

```suggestion
          this.customUserFields.set(customUserFields);